### PR TITLE
update copyright notice

### PR DESCRIPTION
**Pull request by mgovers**
Relates to https://github.com/PowerGridModel/.github/pull/23
Relates to https://github.com/PowerGridModel/power-grid-model-io/pull/218

- Bumps copyright to 
  > 2024 Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
  >
  > SPDX-License-Identifier: MPL-2.0
- Removes Alliander email address in copyright notice (superseded by powergridmodel@lfenergy.org )
- Updates copyright statement in PGM docs
  image
- Updates authors in PyPI/conda
- It's probably a good idea to wait with merge until PGM IO (https://github.com/PowerGridModel/power-grid-model-io/pull/218) and org-level (https://github.com/PowerGridModel/.github/pull/23) copyright statements PRs are also ready

# Script used

I used a simple script to add the new notice:

```py
from pathlib import Path
from subprocess import run, SubprocessError

root = Path(__file__).parent
files = run(["git", "ls-files"], capture_output=True, check=True, encoding="utf-8").stdout.splitlines()
print(files)

failed = []
for f in files:
    command = [
        "reuse",
        "annotate",
        '--copyright=2024 Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>',
        '--license=MPL-2.0',
        "--merge-copyrights",
        f,
    ]
    print(f"updating: '{f}'")
    print(command)
    try:
        run(command, check=True)
    except SubprocessError:
        failed.append(f)

print("failed:")
print(failed)
```
and then manually updated the failing ones.

I then deprecated all `2022` to `2022-2023`

### CODE REVIEW

**Review by Jonasvdbo — CHANGES_REQUESTED** [review]
I noticed that you made a difference with the copyright note for 2022-2023 and 2024. To improve uniformity, clarity and keep it simple, I recommend updating all copyright statements to read: "Contributors to the Power Grid Model Project powergridmodel@lfenergy.org."  For legal purposes, this approach is fine . Moreover, it is acceptable to exclude the specification of years in these statements.

Additionally, for those seeking more information, I suggest directing them to https://alliander.atlassian.net/wiki/spaces/OOSPO/pages/2537140401/Hoe+registreer+ik+de+auteurs+copyright+holders+bij+een+Alliander+OSS-project

**Comment by mgovers** [comment]
> 2022-2024 Contributors to the Power Grid Model Project [powergridmodel@lfenergy.org](mailto:powergridmodel@lfenergy.org)

thank you for the great feedback! will process

File: tests/data/state_estimation/sensor-update/sym_output.json.license
Comment (mgovers) [inline tests/data/state_estimation/sensor-update/sym_output.json.license:1]: `sym_output.json` did not exist (only `sym_output_batch.json` did)

File: tests/data/state_estimation/sensor-update/update_batch.license
Comment (mgovers) [inline tests/data/state_estimation/sensor-update/update_batch.license:1]: file `update_batch` did not exist (only `update_batch.json` did)

File: tests/data/state_estimation/unbalanced-power-measurements/sym_output.json.license
Comment (mgovers) [inline tests/data/state_estimation/unbalanced-power-measurements/sym_output.json.license:1]: file `sym_output.json` did not exist. only `sym_output_batch.json` did

**Review by mgovers — COMMENTED** [review]
I went over all changes again and now everything seems to be consistent

**Comment by mgovers** [comment]
@Jonasvdbo and @TonyXiang8787 for obvious reasons, I would like your formal approval on all PRs before merging them:

* [ ] https://github.com/PowerGridModel/.github/pull/23
* [ ] https://github.com/PowerGridModel/power-grid-model/pull/481
* [ ] https://github.com/PowerGridModel/power-grid-model-io/pull/218
* [ ] https://github.com/PowerGridModel/power-grid-model-workshop/pull/22
* [ ] https://github.com/PowerGridModel/power-grid-model-benchmark/pull/10

**Comment by TonyXiang8787** [comment]
@Jonasvdbo please also approve otherwise we cannot merge.

**Review by Jonasvdbo — APPROVED** [review]

**Pull request merged**